document-store/src/integrationTest/java/org/hypertrace/core/documentstore/postgres/PostgresDocStoreTest.java

```diff
@@ -437,6 +442,120 @@ public void testSearch() throws IOException {
       Assertions.assertTrue(actualMessage.contains(expected));
     }
 
+    // Field exists in the document
+    {
+      Query query = new Query();
+      query.setFilter(new Filter(Op.EXISTS, "testKeyExist", null));
+      Iterator<Document> results = collection.search(query);
+      List<Document> documents = new ArrayList<>();
+      while (results.hasNext()) {
+        documents.add(results.next());
+      }
+      Assertions.assertEquals(2, documents.size());
+    }
+
+    // Nested Field exists in the document
+    {
+      Query query = new Query();
+      query.setFilter(new Filter(Op.EXISTS, "attributes.trace_id.value.testKeyExistNested", null));
+      Iterator<Document> results = collection.search(query);
+      List<Document> documents = new ArrayList<>();
+      while (results.hasNext()) {
+        documents.add(results.next());
+      }
+      Assertions.assertEquals(2, documents.size());
+    }
+
+    // Field Not Exists in the document
+    {
+      Query query = new Query();
+      query.setFilter(new Filter(Op.NOT_EXISTS, "attributes.trace_id.value.testKeyExistNested", null));
+      Iterator<Document> results = collection.search(query);
+      List<Document> documents = new ArrayList<>();
+      while (results.hasNext()) {
+        documents.add(results.next());
+      }
+      Assertions.assertEquals(1, documents.size());
+    }
+
+  }
+
+  @Test
+  public void testExistsFilter() throws IOException {
```

Comment:
Can you add the same test in MongoDocStoreTest?

Reply:
added